### PR TITLE
added MAV_SENSOR_ORIENTATION enum

### PR DESCRIPTION
**Pull request by TSC21**
Rebased to `master`. Sorry :)

### CODE REVIEW

**Comment by TSC21** [comment]
OK seems fine now. Removed a comma that was added by mistake.

**Comment by LorenzMeier** [comment]
Could you rebase interactive and fixup the two commits into one? We try to only put commits onto master which are self-contained. Thanks!

**Comment by TSC21** [comment]
@LorenzMeier you mean redo the PR just with one commit?

**Comment by LorenzMeier** [comment]
https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i/

**Comment by TSC21** [comment]
I done interactive rebase. But how do I push changes?

**Comment by LorenzMeier** [comment]
With --force

**Comment by LorenzMeier** [comment]
Commit history doesn't look good though. It should have a single commit. You need to fetch upstream and rebase against upstream: git rebase upstream/master

**Comment by TSC21** [comment]
Done! Sorry for the time taken

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -1419,6 +1419,126 @@
...
+                  <description>Roll: 270, Pitch: 270, Yaw: 0</description>
+              </entry>
+              <entry value="36" name="MAV_SENSOR_ROTATION_ROLL_90_PITCH_180_YAW_90">
+                  <description>Roll: 90, Pitch: 180, Yaw: 90</description>
+              </entry>
+              <entry value="37" name="MAV_SENSOR_ROTATION_ROLL_90_YAW_270">
+                  <description>Roll: 90, Pitch: 0, Yaw: 270</description>
+              </entry>
+              <entry value="38" name="MAV_SENSOR_ROTATION_MAX">
+                  <description>Roll: 315, Pitch: 315, Yaw: 315</description>
```
Comment (vooon) [inline message_definitions/v1.0/common.xml]: Does it right?

Comment (TSC21) [inline message_definitions/v1.0/common.xml]: Suppose so as we are considering 45 degrees variations.

**Comment by vooon** [comment]
Also does it compatible with https://github.com/PX4/Firmware/blob/master/src/lib/conversion/rotation.h#L50 or https://github.com/diydrones/ardupilot/blob/master/libraries/AP_Math/rotations.h#L28 ?

**Comment by TSC21** [comment]
I used other list of APM firmware (relative to mag as I remember). As you can see it has more types of rotations. So it is compatible with both.

**Comment by TSC21** [comment]
@LorenzMeier is it good to merge? Have one other thing to add, possibly in a next PR.

**Pull request merged**